src/core/VNodes.ts

```diff
@@ -1,20 +1,53 @@
 import {
-	Props,
-	InfernoChildren,
-	VNode,
-	VNodeFlags,
-	createVNode,
-	isVNode,
-} from '../core/shapes';
-import {
 	isArray,
-	isNull,
+	isInvalid,
 	isNullOrUndef,
 	isUndefined,
-	isInvalid
+	isNull,
+	isStatefulComponent
 } from '../shared';
+import {
+	normalize
+} from './normalization';
+import {
+	VNodeFlags,
+	VNode,
+	InfernoChildren,
+	Props,
+	Key,
+	Ref
+} from './structures';
+
+export function createVNode(
+	flags: VNodeFlags,
+	type?,
```

Comment:
This should be `type?: Type` where `Type` is imported from `src/core/structures.ts` for better typings.